### PR TITLE
Features/extend framework

### PR DESCRIPTION
**Pull request by mnishizawa**
**Summary**

**Test plan**

### CODE REVIEW

File: src/core/assertions.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                    currentValue,
+                    `Property '${field}' (${currentValue}) does not contain '${expectedValue}'`
+                ).to.contain(expectedValue)
+                break
+
+            case 'defined':
+            case 'present':
+                expect(currentValue, `Property '${field}' is undefined`).to.not.be.undefined
+                break
+
```
Comment (mnishizawa) [inline src/core/assertions.js]: Lets add notPresent here.

File: src/extensions/http_api/definitions.js [hunk 2]
```diff
@@ -315,7 +315,9 @@ module.exports = ({ baseUrl = '' } = {}) => ({ Given, When, Then }) => {
                 case 'present':
                     expect(currentValue, `Property '${field}' is undefined`).to.not.be.undefined
                     break
-
+                case 'notPresent':
```
Comment (mnishizawa) [inline src/extensions/http_api/definitions.js]: Move to assertions.js

File: src/extensions/http_api/client.js [hunk 3]
```diff
@@ -219,9 +219,9 @@ class HttpApiClient {
             const fullUri = `${baseUrl}${path}`
 
             if (this.body !== null) {
-                if (!['POST', 'PUT'].includes(method)) {
+                if (!['POST', 'PUT', 'PATCH'].includes(method)) {
```
Comment (mnishizawa) [inline src/extensions/http_api/client.js]: Add PATCH to verbsAcceptingBody at the top.

Comment (mnishizawa) [reply src/extensions/http_api/client.js]: We actually shouldn't need the rest of this block as it is done on line 242 below and using the verbsAcceptingBody array from above.

**Review by mFernandezbertocchi — APPROVED** [review]

**Pull request merged**